### PR TITLE
[v2] Update ESLint-related dependencies

### LINKED ISSUE #5411: [v2] Definition for rule 'jsx-a11y/href-no-hash' was not found
## Description

When eslint-plugin-jsx-a11y@6 is installed in a project using gatsby@2, I get the following console warning for every single page:

```
(node:2244) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
 WARNING  Compiled with 1 warnings                                                  19:07:36


C:\Development\Projects\gatsby-v2-test\src\pages\index.jsx
  1:1  warning  Definition for rule 'jsx-a11y/href-no-hash' was not found  jsx-a11y/href-no-hash

✖ 1 problem (0 errors, 1 warning)


You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.
‼ ｢wdm｣:
i ｢wdm｣: Compiled with warnings.
```

### Steps to reproduce

Set up a new gatsby@2 project, run `yarn add -D eslint-plugin-jsx-a11y` and then run `gatsby develop`.

### Environment
* Gatsby version (`npm list gatsby`): 2.0.0-alpha.32
* gatsby-cli version (`gatsby --version`): 2.0.0-alpha.32
* Node.js version: 8.11.1
* Operating System: Windows 10 x64

### File contents (if changed)

`package.json`:

```json
{
  "private": true,
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop"
  },
  "dependencies": {
    "gatsby": "^2.0.0-alpha.32",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "eslint-plugin-jsx-a11y": "^6.0.3"
  }
}
```

### PR DESCRIPTION
**Pull request by kripod**
Fixes #5411

### CODE REVIEW

File: packages/gatsby/package.json [hunk 1]
```diff
@@ -141,6 +141,7 @@
     "@babel/preset-react": "7.0.0-beta.42",
     "@babel/runtime": "7.0.0-beta.42",
     "cross-env": "^5.1.4",
+    "eslint-plugin-react": "^7.8.2",
```
Comment (m-allanson) [inline packages/gatsby/package.json]: This is a duplicate?

File: packages/gatsby/package.json [hunk 2]
```diff
@@ -46,13 +46,13 @@
     "domready": "^1.0.8",
     "dotenv": "^4.0.0",
     "eslint": "^4.19.1",
-    "eslint-config-react-app": "^2.1.0",
+    "eslint-config-react-app": "3.0.0-next.66cc7a90",
```
Comment (m-allanson) [inline packages/gatsby/package.json]: Is this change needed? What is different in `3.0.0` compared to `2.1.0`?

File: packages/gatsby/package.json [hunk 3]
```diff
@@ -46,13 +46,13 @@
...
     "eslint-loader": "^2.0.0",
     "eslint-plugin-flowtype": "^2.46.1",
     "eslint-plugin-graphql": "^2.0.0",
     "eslint-plugin-import": "^2.9.0",
-    "eslint-plugin-jsx-a11y": "^5.1.1",
-    "eslint-plugin-react": "^7.7.0",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
```
Comment (m-allanson) [inline packages/gatsby/package.json]: 👍 this is the fix, isn't it?

File: packages/gatsby/package.json (hunk 2 above)
Comment (kripod) [reply packages/gatsby/package.json]: Yeah, it’s necessary as 2.1.0 is made using eslint-plugin-jsx-a11y@5, while 3.0.0 is made using eslint-plugin-jsx-a11y@6

Comment (kripod) [reply packages/gatsby/package.json]: 2.1.0 refers a deprecated rule from eslint-plugin-jsx-a11y@5

File: packages/gatsby/package.json (hunk 1 above)
Comment (kripod) [reply packages/gatsby/package.json]: It seems so. `yarn upgrade-interactive` seems to have played tricks on me. 😅

File: packages/gatsby/package.json (hunk 3 above)
Comment (kripod) [reply packages/gatsby/package.json]: As I have mentioned before, eslint-config-react-app@2.1.0 refers to a deprecated rule (`jsx-a11y/href-no-hash`) from eslint-plugin-jsx-a11y@5, which has been replaced (with `jsx-a11y/anchor-is-valid`) in eslint-plugin-jsx-a11y@6.

File: packages/gatsby/package.json (hunk 2 above)
Comment (m-allanson) [reply packages/gatsby/package.json]: Makes sense, thank you.

File: packages/gatsby/package.json (hunk 3 above)
Comment (m-allanson) [reply packages/gatsby/package.json]: 👍
